Plutarch/Lift.hs

```diff
@@ -41,7 +50,7 @@ Laws:
    in UPLC (via the 'UntypedPlutusCore.Constant' constructor) is a valid @p@.
 -}
 class (PConstant (PLifted p), PConstanted (PLifted p) ~ p) => PUnsafeLiftDecl (p :: PType) where
-  type PLifted p :: Type
+  type PLifted p = (r :: Type) | r -> p
```

Comment:
The tests were erroring out since they had trouble figuring out the transitive `PLifted p` result in the newly added `PLift` type alias- `type PLift p = (p ~ PConstanted (PLifted p), PUnsafeLiftDecl p)` - only seemed to happen in `pconstant` usages.

This type family dependency fixes that and helps GHC immediately figure out the transitive result.

I honestly think we should sprinkle in type family deps everywhere, but especially across the `PConstant`/`PLift` architecture - I quite like type fam deps. It should be theoretically possible but our abstract derivers make it extremely complicated to _not_ violate the injectivity principles if adding type fam deps to the `PConstant` associated types.

Reply:
I hadn't realised this was possible for associated type families! This seems quite powerful.